### PR TITLE
feat: add yaml config file

### PR DESCRIPTION
**Pull request by watermelon1024**
# Description

Add config file (yaml).

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x]  This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: config.yml.example [hunk 1]
```diff
@@ -0,0 +1,12 @@
+dvc_id:
+  - int  # 創建動態語音 頻道ID
+  - ...  # 可多個，以此類推
+member_role_id: int  # @成員 身份組ID
```
Comment (SiongSng) [inline config.yml.example:4]:
```suggestion
member_role_id: int  # @組織成員 身份組ID
```

**Review by banahaker — APPROVED** [review]
Nice Commit

File: core/bot.py [hunk 2]
```diff
@@ -45,8 +46,15 @@ def __init__(self, *args, **kwargs):
             'newApplyServerToken': os.getenv('NEW_APPLY_SERVER_TOKEN'),
         }
 
-        # dynamic voice channel ID
-        self.dvc_ids: List[int] = [977050757461336124]
+        # config setting
```
Comment (SiongSng) [inline core/bot.py:49]: config setting 🤔

File: core/bot.py [hunk 3]
```diff
@@ -45,8 +46,15 @@ def __init__(self, *args, **kwargs):
...
+        with open("config.yml", "r", encoding="utf8") as config_yaml:
+            config = yaml.load(config_yaml, yaml.Loader)
+        self.log.info("load config.yml is complete")
+        # print(config)
```
Comment (SiongSng) [inline core/bot.py:53]:
```suggestion

```
請刪除不必要的程式碼

File: core/bot.py [hunk 4]
```diff
@@ -45,8 +46,15 @@ def __init__(self, *args, **kwargs):
...
+        self.dvc_ids: List[int] = config['dvc_id']
```
Comment (SiongSng) [inline core/bot.py:54]: config 中是 `dvc_id` 在程式碼中卻是 `dcv_ids`，請保持一致性

File: README.md [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 the bot of Lipoic official discord server
 
+Please view `config.yml.example` and create a `config.yml` file.
```
Comment (SiongSng) [inline README.md:5]:
```suggestion
Please follow the format of `config.yml.example` to create a `config.yml` file.
```

這樣應該比較符合這段意思，請按照 `config.yml.example` 的格式來建立 `config.yml` 檔案。

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: cogs/admin/mute.py [hunk 6]
```diff
@@ -4,9 +4,13 @@
...
 import datetime
 
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from core import LIPOIC
+
 
 class MuteCog(discord.Cog):
-    def __init__(self, bot):
+    def __init__(self, bot: 'LIPOIC'):
```
Comment (SiongSng) [inline cogs/admin/mute.py:13]:
```suggestion
    def __init__(self, bot: 'Lipoic'):
```

File: cogs/admin/pin.py [hunk 7]
```diff
@@ -1,22 +1,37 @@
...
 import datetime
 
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from core import LIPOIC
+
 
 class PinCog(discord.Cog):
-    def __init__(self, bot):
+    def __init__(self, bot: 'LIPOIC'):
```
Comment (SiongSng) [inline cogs/admin/pin.py:13]:
```suggestion
    def __init__(self, bot: 'Lipoic'):
```

**Review by SiongSng — CHANGES_REQUESTED** [review]
Lipoic 通常不會大寫

**Pull request merged**